contentcuration/contentcuration/frontend/channelEdit/vuex/task/index.js

```diff
@@ -47,14 +47,24 @@ export default {
       return Task.deleteModel(task.task_id).then(() => store.dispatch('activateTaskUpdateTimer'));
     },
     updateTaskList(store) {
-      return Task.where({ channel: store.rootState.currentChannel.currentChannelId })
-        .then(tasks => {
-          store.commit('SET_ASYNC_TASKS', tasks);
-          store.dispatch('activateTaskUpdateTimer');
-        })
-        .catch(() => {
-          store.dispatch('activateTaskUpdateTimer');
-        });
+      if (store.rootGetters.loggedIn) {
+        return Task.where({ channel: store.rootState.currentChannel.currentChannelId })
+          .then(tasks => {
+            store.commit('SET_ASYNC_TASKS', tasks);
+            store.dispatch('activateTaskUpdateTimer');
+          })
+          .catch(e => {
+            // if not authorized, redirect to login screen
+            if (e.response.status === 403) {
+              this.$router.push('/');
+            }
+            store.dispatch('activateTaskUpdateTimer');
+          });
+      } else {
+        //  for cases where this.$router doesn't exist when user is signed out,
+        // use window.Urls to redirect to login
+        window.location.href = window.Urls.login();
```

Comment:
Did `import urls from 'shared/urls';` not work here? Allows us to decouple from a specific name on the window object.

Reply:
It does work, I just didn't retain that suggestion despite reading it. Updated!